### PR TITLE
Add resource for enabling firebase

### PR DESCRIPTION
**Pull request by chrisst**
Also adding a new feature to generated tests to not attempt to validate resources with CheckDestroy if they aren't deletable. 

My testing so far as been with service accounts and, unless my environment is leaking personal credentials, the API is tolerating it just fine. Even though the official Firebase documentation contradicts this.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
firebase: Add the `google_firebase_project` resource which will enable Firebase for a referenced Google project.
```

PR for https://github.com/terraform-providers/terraform-provider-google/issues/2973

### CODE REVIEW

File: products/firebase/api.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      Since a FirebaseProject is actually also a GCP Project, a FirebaseProject uses underlying GCP
+      identifiers (most importantly, the projectId) as its own for easy interop with GCP APIs.
+
+      Once Firebase has been added to a Google Project it cannot be removed.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/'
+      api: 'https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects'
+    properties:
```
Comment (chrisst) [inline products/firebase/api.yaml]: There are a couple other properties that come back in the response, but they all appear to have the same value and all of them are the project-id. So I wasn't sure if it was worth adding them.

ex:
```
{
  "projectId": "tf-testjeyulibspx",
  "projectNumber": "684476938629",
  "displayName": "tf-testjeyulibspx",
  "name": "projects/tf-testjeyulibspx",
  "resources": {
    "hostingSite": "tf-testjeyulibspx",
    "realtimeDatabaseInstance": "tf-testjeyulibspx"
  }
}
```

**Comment by chrisst** [comment]
k - done cleaning up this PR and ready for a review. Sorry about the churn.

File: provider/terraform/examples.rb [hunk 2]
```diff
@@ -107,6 +107,8 @@ class Examples < Api::Object
       # Whether to skip generating tests for this resource
       attr_reader :skip_test
 
+      attr_reader :skip_delete
```
Comment (ndmckinley) [inline provider/terraform/examples.rb]: Is this skipping all deletes, or just skipping the test for deletion?

Comment (chrisst) [reply provider/terraform/examples.rb]: This only skips the delete call in the generated test. The [custom delete](https://github.com/GoogleCloudPlatform/magic-modules/pull/3281/files#diff-6a7362a33a2f6b7a6896d550919a49a4R24) is what actually stops the resource from deleting.

Comment (ndmckinley) [reply provider/terraform/examples.rb]: Could probably use a different name, or a comment maybe, or else you could genericize the custom delete?

Comment (ndmckinley) [reply provider/terraform/examples.rb]: Oh, looks like it's already pretty generic.  you could maybe make this automatically add the custom delete and move the existing code around a bit to make it consistent?

**Comment by chrisst** [comment]
ah yeah, update coming.

**Comment by alixaxel** [comment]
Still no Firestore provisioning support right?

**Comment by chrisst** [comment]
Correct

**Pull request merged**